### ISSUE TITLE
Toggl private projects stop import process

### ISSUE THREAD

**Issue by ricardorover** [mikerourke/transfermyti.me#32]
Hi, thanks for the nice tool!

I noticed when importing projects list from Toggl to Clockify, that if in Toggl there is a project as "Private" and with some users selected to see it, the transfermyti.me tool stops Projects importing due to a "400" response status from Clockify.

I imagine Clockify tries to find the user to add the correct permission aswell, but in my scenario the user does not exist.

In my scenario there is no need for private projects, so I'm changing all my Toggl projects to be not Private, but took a while to understand where was the problem.

Not sure how transfermyti.me could handle it, maybe handling the error to warn the user about this is good enough for now.

**Comment by mikerourke** [comment]
Hi @ricardorover, you're very welcome! I apologize for the issue. I'll try recreating it in my test account and add some accommodations to the code. I started working on adding user/user group transfer to the app, but that'll be tied pretty heavily to organizational/team transfer. I plan on adding that in the future, but it's going to be a pretty massive undertaking, so it's commented out for now.

I'm not sure if the 400 response coming back from Clockify will actually tell me if the transfer error was due to this specific scenario, but I could try checking the data beforehand and display a warning to the user. Thanks for bringing this to my attention. Some bugs have been popping up due to situations I didn't accommodate for in development, and I'm trying to make the transfer process as smooth as possible. I'll let you know when the issue is fixed. Thanks!

**Comment by mikerourke** [comment]
Hey @ricardorover, so I tried the following and I wasn't able to recreate the error:

1. Created new testing Toggl account with user name: `testmember`
2. Invited `testmember` to my original testing Toggl account (user name: `testing`)
3. Assigned `testmember` to one of my projects as a non-manager (all of my Toggl projects are private)
4. Transferred the project from Toggl to Clockify using transfermyti.me.

Everything worked as expected. I also deleted the project on Clockify, removed `testmember` from the project, re-added them with **Manager** access and tried it again and it worked. Was anything else failing, or was it just the projects?

**Comment by ricardorover** [comment]
Hey @mikerourke, thanks for the fast action, and sorry for my delay

I was having a bad time trying to reproduce as well, until I remembered the first tries I was importing several Toggl workspaces at the same time. I believe this caused the problem, because I tried the same operation to a new Clockify account and the error occured.

I copied the request that returned 400 status as cURL:
```
curl 'https://api.clockify.me/api/v1/workspaces/5e4def3ce4acab6e6c0fc5c6/projects' -H 'Connection: keep-alive' -H 'DNT: 1' -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36' -H 'X-Api-Key: <MY_SECRET_CLOCKIFY_API_KEY>' -H 'Content-Type: application/json' -H 'Accept: */*' -H 'Origin: https://transfermyti.me' -H 'Sec-Fetch-Site: cross-site' -H 'Sec-Fetch-Mode: cors' -H 'Referer: https://transfermyti.me/perform-tool-action' -H 'Accept-Encoding: gzip, deflate, br' -H 'Accept-Language: pt-BR,pt;q=0.9,en-US;q=0.8,en;q=0.7,it-IT;q=0.6,it;q=0.5,pt-PT;q=0.4' --data-binary '{"name":"Loeffa-Admin","clientId":"5e4df552e4acab6e6c0fcc2f","isPublic":false,"estimate":{"estimate":0,"type":"MANUAL"},"color":"#e19a86","billable":false}' --compressed
```

Running this at command line got the server response:
```
{"message":"CLIENT with id 5e4df552e4acab6e6c0fcc2f doesn't belong to WORKSPACE with id 5e4def3ce4acab6e6c0fc5c6","code":501}
```

I could reproduce the error having the same client name in 2 Toggl workspaces (newly created), and having a project for each client in its workspace. Not sure it was the first scenario, but it's a very similar error.

So, it's not related to private project setting, as I first imagined, so I believe makes sense changing the issue's title.

Hope it helps!

PS: Sorry for suggesting the wrong cause!

**Comment by mikerourke** [comment]
No problem, thanks for getting back to me. So I'm currently not accommodating for the same client/tag/project name existing in 2 workspaces. Right now I just check if the names match and use that for linking. The tool tries using an invalid ID for the workspace and errors out. I'll update the code to ensure the linking is based on the name **and** workspace. I'll let you know when it's fixed. Good catch!

**Comment by mikerourke** [comment]
FYI, @ricardorover, I believe I fixed the issue on #36. I ran some tests and it looks like everything is working. Let me know if you're still having issues. Thanks!
